mevislab.github.io/content/tutorials.md

```diff
@@ -0,0 +1,225 @@
+| `.script` | `MDL` script file, typically includes the user interface definition for panels. See [Chapter GUI Development](./tutorials/macromodules/guidesign#Example_Paneldesign "GUI Development") for an example on GUI programming. |
+| `.mlimage` | MeVisLab internal image format for 6D images saved with all DICOM tags, lossless compression, and in all data types. |
+| `.mhelp` | File with descriptions of all fields and the use of a module, edit- and creatable by using `MATE`. See [Help files](./tutorials/macromodules/helpfiles "Help files") for details. |
+| `.py` | Python file, used for scripting in macro modules. See [Python scripting](./tutorials/macromodules/pythonscripting#TutorialPythonScripting "Python scripting") for an example on macro programming. |
+| `.dcm` | DCM part of the imported DICOM file, see [Importing DICOM Data](./tutorials/basicmechanisms/dataimport#DICOMImport "Importing DICOM Data"). |
+
+### Types of Modules {#Module_Types}
+
+Within the concept of MeVisLab the basic entities we are working with
+a graphical representations of modules having their specific functions
+for image processing, image visualization, and image interaction.
+
+The three basic module types (ML, Open Inventor and Macro Module) are
+distinguished by their colors:
+
+| <div style="width:230px">Type</div> | <div style="width:430px">Look</div> | Characteristics |
+| --- | --- | --- |
+| ML Module (blue) | ![ML Module](/images/MLMModuleML.png "ML Module") | Page-based, demand-driven processing of voxels |
+| Open Inventor Module (green) | ![Open Inventor Module](/images/MLMModuleSo.png "Open Inventor Module") | Visual scene graphs (3D); naming convention: all modules starting with *So* (for **s**cene **o**bject) |
+| Macro Module (brown) | ![Macro Module](/images/MLMModuleMacro.png "Macro Module") | Combination of other module types, allowing implementing hierarchies and scripted interaction |
+
+If a module is invalid, it is displayed in bright red. This might happen if the module itself is not available on your system.
+
+
+| <div style="width:330px">Appearance</div> | Explanation |
+| --- | --- |
+| ![Invalid Module](/images/MLMModuleStateInvalid.png "Invalid Module") | Invalid module |
+  ![Macro State Invalid](/images/MLMModuleStateMacroInvalidModule.png "Macro State Invalid") | Macro containing an invalid module |
+
+The number of warning and error messages that are being printed to the
+debug console are listed at the upper right corner of the module. Once
+the debug console is cleared, the warning and error indicators at the
+module are also cleared. If the module produces information messages,
+their number is printed in gray at this position. This enables a network
+or module developer to find the modules in a network that produce
+messages quickly.
```

Comment:
This allows a network or module developer to quickly find the modules in a network that produce
messages.